ban/core/versioning.py

```diff
@@ -277,16 +270,51 @@ def serialize(self, *args):
         }
 
 
-class IdentifierRedirect(db.Model):
+class Redirect(db.Model):
+
     model_name = db.CharField(max_length=64)
     identifier = db.CharField(max_length=64)
-    old = db.CharField(max_length=255)
-    new = db.CharField(max_length=255)
+    value = db.CharField(max_length=255)
+    to = db.CharField(max_length=255)
+
+    class Meta:
+        primary_key = peewee.CompositeKey('model_name', 'identifier', 'value',
+                                          'to')
+
+    @classmethod
+    def add(cls, instance, identifier, value):
+        if isinstance(instance, tuple):
+            # Optim so we don't need to request db when creating a redirect
+            # from a diff.
+            model_name, to = instance
+        else:
+            model_name = instance.__class__.__name__
```

Comment:
Not fond of these `.__class__.__name__` everywhere, why not a `model_name` property at the instance level?